aten/src/ATen/core/ivalue.h

```diff
@@ -248,14 +248,22 @@ struct CAFFE2_API IValue final {
   // While some of these accessors could be generated through templates,
   // we prefer to write them manually for clarity
 
-  IValue(at::Tensor t)
-  : tag(Tag::Tensor), is_intrusive_ptr(t.defined())  {
-    // Note: the undefined tensor is not refcounted, so while it
-    // is tagged as a tensor, is_intrusive_ptr is set to false.
-    // This is not an optional optimization: our incref call
-    // *will not* do the right thing when called on an
-    // undefined tensor.
-    payload.as_intrusive_ptr = t.unsafeReleaseTensorImpl();
+  IValue(at::Tensor t) {
+    if (t.defined()) {
+      tag = Tag::Tensor;
+      is_intrusive_ptr = true;
+      // Note: the undefined tensor is not refcounted, so while it
+      // is tagged as a tensor, is_intrusive_ptr is set to false.
+      // This is not an optional optimization: our incref call
+      // *will not* do the right thing when called on an
+      // undefined tensor.
+      payload.as_intrusive_ptr = t.unsafeReleaseTensorImpl();
+    } else {
+      // If the tensor was undefined, set the IValue to None instead
```

Comment:
This means that undefined tensors no longer return true for `isTensor()`, which seems a little confusing:
```
auto iv = IValue(at::Tensor())
iv.isTensor()  // false???
```
This may also have wide-reaching implications. For example, in `constant_propagation.cpp`, I see code that looks like:
```
    if (v.isTensor()) {
      at::Tensor t = v.toTensor();
      if (t.defined() && t.requires_grad()) {
        ...
```
with this change, the initial check for `isTensor` will fail, causing the whole block to to be skipped. I'm not sure if it matters, but we'll have to check all the callsites for `isTensor` to make sure they do not care about this change.

Reply:
ugh. How much do you think we can rely on CI here?